api/server.go

```diff
@@ -41,13 +41,22 @@ type octetResponse struct {
 	Reader io.Reader
 }
 
-type handler func(http.ResponseWriter, *http.Request) *errorResponse
+type handler struct {
+	fn     func(http.ResponseWriter, *http.Request) *errorResponse
+	record bool
```

Comment:
This is hiding too much, because most things don't set it (so it defaults to false). Going to switch to something that you can opt-in to.